workspace_host/interface.js

```diff
@@ -719,3 +734,51 @@ function destroySequence(workspace_id, res) {
         }
     });
 }
+
+function gradeSequence(workspace_id, res) {
+    var workspace_dir = `${workspacePrefix}/${id_workspace_mapper[workspace_id].localName}/project`;
+    var graded_file_list = id_workspace_mapper[workspace_id].settings.workspace_graded_files.map((file) => {return path.join(workspace_dir, file);});
+    graded_file_list = ['starter_code.c', 'starter_code.h'].map((file) => {return path.join(workspace_dir, file);});    // since in the DB the workspace_graded_file col is empty, hardcode file list
+    console.log(graded_file_list);
+    var timestamp = new Date().toISOString().replace(/T/, '*').replace(/\..+/, '');
+    var zip_file_name = `workspace-${workspace_id}-${timestamp}.tgz`;
+
+    async.series([
+        async () => {
+            for (var file_path in graded_file_list) {
+                try {
+                    await fsPromises.lstat(file_path);
+                } catch(err) {
+                    return(`Required graded file ${path.basename(file_path)} does not exist.`);
```

Comment:
Rather than returning an error if the file is missing, I think it would be better to return all the files we do have (possibly zero) and let the PL server figure out what to do with errors.

Reply:
Done. And I modified the code so it doesn't contain the unnecessary folder in the zip file.